lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2041,3 +2041,10 @@ bool BuildSystem::enableTracing(StringRef path,
 bool BuildSystem::build(StringRef name) {
   return static_cast<BuildSystemImpl*>(impl)->build(name);
 }
+
+void BuildSystem::cancel() {
+  if (impl) {
```

Comment:
I moved up the flag from the C API's delegate and merged it with the existing `isCancelled()` implementation in the frontend delegate class.